runtime/src/snapshot_utils.rs

```diff
@@ -1998,6 +2017,29 @@ mod tests {
         assert_matches!(result, Err(SnapshotError::Io(ref message)) if message.to_string().starts_with("invalid snapshot data file"));
     }
 
+    #[test]
+    fn test_snapshot_version_from_file_under_limit() {
+        let file_content = format!("v{}", DEFAULT_SNAPSHOT_VERSION);
+        let mut file = NamedTempFile::new().unwrap();
+        file.write_all(file_content.as_bytes()).unwrap();
+        let version_from_file = snapshot_version_from_file(file.path()).unwrap();
+        assert_eq!(version_from_file, file_content);
+    }
+
+    #[test]
+    fn test_snapshot_version_from_file_over_limit() {
+        // Verify `as usize` below is safe.
+        usize::try_from(MAX_SNAPSHOT_VERSION_FILE_SIZE + 1).unwrap();
```

Comment:
Capture this as a variable and use it for the size in the `file_content` array below.